assignments/02-collections-lifetimes/src/life.rs

```diff
@@ -1,19 +1,100 @@
-fn split_string(string: &str, delimeter: &str) -> Vec<&str> {
-    todo!()
+fn split_string<'a>(string: &'a str, delimiter: &str) -> Vec<&'a str> {
+    let mut split_string: Vec<&str> = Vec::new();
+    let mut start = 0;
+
+    while start < string.len() {
+        if let Some(pos) = string[start..].find(delimiter) {
+            split_string.push(&string[start..start + pos]);
+            start += pos + delimiter.len();
+        } else {
+            split_string.push(&string[start..]);
+            break;
+        }
+    }
+
+    split_string
 }
 
 #[derive(PartialEq, Debug)]
-struct Differences {
-    only_in_first: Vec<&str>,
-    only_in_second: Vec<&str>,
+struct Differences<'a> {
+    only_in_first: Vec<&'a str>,
+    only_in_second: Vec<&'a str>,
```

Comment:
This works, but they don't necessarily have to be in the same lifetime (one can be `'a` & another `'b`